tests/integration/test_external_access.py

```diff
@@ -79,22 +77,21 @@ def test_nodeport_service(
 
     assert_service_status(namespace=cast(str, juju.model), service_type="NodePort")
 
-    username = "admin"
-    password = fetch_password(juju)
+    jdbc_endpoint, username, password = fetch_connection_info(
+        juju, charm_versions.data_integrator.app
+    )
 
     # Run SQL tests against JDBC endpoint
-    jdbc_endpoint = fetch_jdbc_endpoint(juju)
     assert run_sql_test_against_jdbc_endpoint(
         juju, test_pod=test_pod, jdbc_endpoint=jdbc_endpoint, username=username, password=password
     )
     assert is_entire_cluster_responding_requests(
-        juju, test_pod=test_pod, username=username, password=password
+        juju, test_pod=test_pod, jdbc_endpoint=jdbc_endpoint, username=username, password=password
```

Comment:
in reference to our discussion earlier about returning multivalued endpoints / uris -- in that case here we won't be able to stuff in the endpoint we receive from data-integrator. The client needs to properly parse, split and choose among the multiple endpoints.